crates/bitwarden-core/src/client/internal.rs

```diff
@@ -223,30 +210,15 @@ impl InternalClient {
 
     #[cfg(feature = "secrets")]
     pub(crate) fn initialize_crypto_single_key(&self, key: SymmetricCryptoKey) {
-        *self
-            .encryption_settings
-            .write()
-            .expect("RwLock is not poisoned") =
-            Some(Arc::new(EncryptionSettings::new_single_key(key)));
+        EncryptionSettings::new_single_key(key, &self.key_store);
     }
 
     #[cfg(feature = "internal")]
     pub fn initialize_org_crypto(
         &self,
         org_keys: Vec<(Uuid, AsymmetricEncString)>,
-    ) -> Result<Arc<EncryptionSettings>, EncryptionSettingsError> {
-        let mut guard = self
-            .encryption_settings
-            .write()
-            .expect("RwLock is not poisoned");
-
-        let Some(enc) = guard.as_mut() else {
-            return Err(VaultLockedError.into());
-        };
-
-        let inner = Arc::make_mut(enc);
-        inner.set_org_keys(org_keys)?;
-
-        Ok(enc.clone())
+    ) -> Result<(), EncryptionSettingsError> {
+        EncryptionSettings::set_org_keys(org_keys, &self.key_store)?;
+        Ok(())
```

Comment:
We can just rely on the return type from `set_org_keys`.
```suggestion
        EncryptionSettings::set_org_keys(org_keys, &self.key_store)
```